lib/api/funnel.js

```
@@ -753,11 +753,29 @@ class Funnel {
    * @returns {Promise}
    */
   async executePluginRequest(request) {
-    try {
-      if (request.input.triggerEvents) {
-        const response = await this.processRequest(request);
-        return { ...response.result };
+    if (request.input.triggerEvents) {
+      try {
+        const { result } = await this.processRequest(request);
+        debug(
+          "Request %s successfully executed. Result: %a",
+          request.id,
+          result,
+        );
+        return global.kuzzle.pipe("request:afterExecution", {
+          request,
+          result,
+          success: true,
+        });
+      } catch (error) {
+        debug("Error processing request %s: %a", request.id, error);
+        return global.kuzzle.pipe("request:afterExecution", {
```

We need to rethrow the error after the error event trigger. If we let this as it, this will change the final pipe error behavior which results in a hidden edge effect/breaking in pipe development experience. You should add a note to update the behavior on next major since we should do as it on next major

---

After checking the behaviour without the triggerEvent flag it appears that any error inside the pipe is caught
For this reason we shouldn't rethrow the error to have the same behaviour when the flag is enabled